topic-IntroOOP

bohledevs opened the pull request:
Finished the 4 Exercises of the topic intro to OOP.

Exercises contain:
- .java's 
- README.md
- UML Model

Consider them highly generalized, as my knowledge of Query and DB is yet shallow.

Commit:
Added the Singleton DB Connection

Commit:
Officially finished the Singleton DB Exercise

Commit:
Partially Finished the Abstract Factory design

Commit:
Finished the Proxy Pattern Exercise

Commit:
Added a README.MD to Abstract Factory project

Commit:
Finished the Builder -  four Exercises completed

LeoBur commented on IntroOOP/AbstractFactory/src/com/globant/abstractfactory/connection/SQLConnection.java:
```diff
@@ -0,0 +1,8 @@
+public interface SQLConnection {
+
+// All connections must establish a session with its respective server.
+  public void connect();
```
Access modifier is redundant

LeoBur commented on IntroOOP/AbstractFactory/src/com/globant/abstractfactory/factory/FactoryProducer.java:
```diff
@@ -0,0 +1,13 @@
+public class FactoryProducer {
+
+  public static SQLFactory getFactory(String factoryType) {
+    if (factoryType.equalsIgnoreCase("MySQL")) {
```
This could throw a NullPointerException(NPE). Check [yoda conditions](https://knowthecode.io/yoda-conditions-yoda-not-yoda) to avoid this

LeoBur commented on IntroOOP/AbstractFactory/src/com/globant/abstractfactory/factory/FactoryProducer.java:
```diff
@@ -0,0 +1,13 @@
+  public static SQLFactory getFactory(String factoryType) {
+    if (factoryType.equalsIgnoreCase("MySQL")) {
+      return new MySQLFactory();
+    } else if (factoryType.equalsIgnoreCase("PostgreSQL")) {
```
The same here

LeoBur commented on IntroOOP/Builder/src/com/globant/builder/connection/Password.java:
```diff
@@ -0,0 +1,15 @@
+
+public class Password extends ConnectionItem {
+  @Override
+  public String getValue() {
```
I see this block of code repeated many times. Refactor the code to avoid this

bohledevs replied:
Ok. I will! 
Thanks.

LeoBur reviewed (commented):
Review the builder pattern documentation and your implementation

Commit:
Refactored code to Yoda conditions

Commit:
Refactorized the method getValue() for connection items in  Builder project

Leandromaro commented on IntroOOP/AbstractFactory/src/com/globant/abstractfactory/app/SQLFactoryDemo.java:
```diff
@@ -0,0 +1,41 @@
+import com.globant.abstractfactory.factory.FactoryProducer;
+import com.globant.abstractfactory.factory.SQLFactory;
+import java.util.logging.Logger;
+
```
Comentarios

bohledevs replied:
Hello Leandro, already added the javaDocs on [this commit](https://github.com/java-bootcamp-rcia/facundo-bohle/pull/1/commits/e59083731957e0c3f9ffeb0d22d0f9a02f757b45)

Commit:
Topic introOOP - Refactored the code
- Added JavaDocs to all classes (except tess) and public methods
- Deleted most in-line comments, except those necessary on application classes
- Modified method return values to simplify the Mock
